packages/runtime/container-runtime/src/test/dataStoreContext.spec.ts

```diff
@@ -380,10 +384,53 @@ describe("Data Store Context Tests", () => {
             });
 
             it("can correctly initialize and generate attributes without version and isRootDataStore", async () => {
+                dataStoreAttributes = { pkg: "TestDataStore1" };
+                const buffer = stringToBuffer(JSON.stringify(dataStoreAttributes), "utf8");
+                const blobCache = new Map<string, ArrayBufferLike>([["fluidDataStoreAttributes", buffer]]);
+                const snapshotTree: ISnapshotTree = {
+                    blobs: { [dataStoreAttributesBlobName]: "fluidDataStoreAttributes" },
+                    commits: {},
+                    trees: {},
+                };
+
+                remotedDataStoreContext = new RemotedFluidDataStoreContext(
+                    dataStoreId,
+                    snapshotTree,
+                    containerRuntime,
+                    new BlobCacheStorageService(storage as IDocumentStorageService, blobCache),
+                    scope,
+                    createSummarizerNodeFn,
+                );
+
+                const isRootNode = await remotedDataStoreContext.isRoot();
+                assert.strictEqual(isRootNode, true, "The data store should be root.");
+
+                const summarizeResult = await remotedDataStoreContext.summarize(true /* fullTree */);
+                assert(summarizeResult.summary.type === SummaryType.Tree,
+                    "summarize should always return a tree when fullTree is true");
+                const blob = summarizeResult.summary.tree[dataStoreAttributesBlobName] as ISummaryBlob;
+
+                const contents = JSON.parse(blob.content as string) as WriteFluidDataStoreAttributes;
+                assert.strictEqual(
+                    contents.pkg,
+                    JSON.stringify([dataStoreAttributes.pkg]),
+                    "Remote DataStore package does not match.");
+                assert.strictEqual(
+                    contents.summaryFormatVersion,
+                    2,
+                    "Remote DataStore snapshot version does not match.");
+                // Remote context without the isRootDataStore flag in the snapshot should default it to true.
+                assert.strictEqual(contents.isRootDataStore, true, "Remote DataStore root state does not match.");
+            });
+
+            it("can correctly initialize and generate attributes with isolated channels disabled", async () => {
                 dataStoreAttributes = {
-                    pkg: "TestDataStore1",
-                    snapshotFormatVersion: undefined,
+                    pkg: JSON.stringify(["TestDataStore1"]),
+                    summaryFormatVersion: 2,
+                    isRootDataStore: true,
                 };
+                (containerRuntime as any).disableIsolatedChannels = true;
```

Comment:
You should set this in the containerRuntime creation in the beforeEach block. And does it make sense to have tests with this set to false?

Reply:
Moved it to a function.